src/devices/SocketCan/CanFrame.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Iot.Device.SocketCan
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct CanFrame
```

Comment:
>CanFrame [](start = 25, length = 8)

would make sense to have a constructor for this struct which take the RawId? I think this struct is useless without RawId. And may be the RawId property be getter only?

Reply:
one of the expected scenarios is that ID changes frequently and the same message might be sent to multiple people - I think it makes sense to leave this mutable

Reply:
talked offline - I will create CanId struct and make CanFrame internal and have WriteFrame and TryReadFrame methods instead

Reply:
struct's still mutable but I've made RawId internal now. Standard and Extended address can still be changed